src/Models/User.php

```diff
@@ -66,6 +68,10 @@ protected static function booted(): void
                 $user->iban = str_replace(' ', '', strtoupper($user->iban));
             }
         });
+
+        static::saved(function (User $user) {
+            Cache::forget('morph_to:' . $user->getMorphClass() . ':' . $user->id . ':name');
```

Comment:
In theory all morph_to:user:id cache keys have to be forgotten, not only :name. Also :name must only be forgotten if it got changed.